tests/phpunit/api/v3/EckEntityTypeTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+use Civi\Test\HeadlessInterface;
+use Civi\Test\HookInterface;
+use Civi\Test\TransactionalInterface;
+
+/**
+ * EckEntityType API Test Case
+ * @group headless
+ */
+class api_v3_EckEntityTypeTest extends \PHPUnit\Framework\TestCase implements HeadlessInterface, HookInterface {
+  use \Civi\Test\Api3TestTrait;
+
+  protected $_apiversion = 3;
+
+  /**
+   * Set up for headless tests.
+   *
+   * Civi\Test has many helpers, like install(), uninstall(), sql(), and sqlFile().
+   *
+   * See: https://docs.civicrm.org/dev/en/latest/testing/phpunit/#civitest
+   */
+  public function setUpHeadless() {
+    return \Civi\Test::headless()
+      ->installMe(__DIR__)
+      ->apply();
+  }
+
+  /**
+   * Test basic APIv3 functionality
+   */
+  public function testCreateGetDelete() {
+    $entityType = $this->callAPISuccess('EckEntityType', 'create', [
+      'label' => 'TestAPIv3',
+    ]);
+    $this->assertTrue(is_numeric($entityType['id']));
+
+    $get = $this->callAPISuccess('EckEntityType', 'get', ['name' => 'TestAPIv3']);
+    $this->assertEquals(1, $get['count']);
+
+    // Table should have been created
+    $this->assertTrue(\CRM_Core_DAO::checkTableExists('civicrm_eck_testapiv3'));
+
+    $records = $this->callAPISuccess('EckTestAPIv3', 'get');
```

Comment:
You can run this test on a previous revision prior to #20 and it will get this far, but then fail on the `create` api call below. That's why I think `create` was always broken.